tools/gyp/pylib/gyp/xcode_emulation.py

```diff
@@ -949,6 +949,8 @@ def GetLibtoolflags(self, configname):
     libtoolflags = []
 
     for libtoolflag in self._Settings().get('OTHER_LDFLAGS', []):
+      if libtoolflag == "--coverage":
```

Comment:
Might I suggest
```diff
diff --git a/node.gypi b/node.gypi
index 0b27e6ad2c..169ca41255 100644
--- a/node.gypi
+++ b/node.gypi
@@ -275,9 +275,6 @@
                    '-O0' ],
       'cflags!': [ '-O3' ],
       'xcode_settings': {
-        'OTHER_LDFLAGS': [
-          '--coverage',
-        ],
         'OTHER_CFLAGS+': [
           '--coverage',
           '-g',
```
instead?

Reply:
Ok after some digging is seem like the native `libtool` does not accept any flag that starts with `--`
Can I suggest this instead of the whole for expression (From L948 to L955)
```python
    self.configname = configname
    libtoolflags = self._Settings().get('OTHER_LDFLAGS', [])
    # Native macOS libtool does not except any flags with '--' prefix
    libtoolflags = [f for f in libtoolflags if not f.startswith('--')]
    # TODO(thakis): ARCHS?
```